tools/c7n_gcp/c7n_gcp/filters/metrics_filter.py

```diff
@@ -0,0 +1,273 @@
+# Copyright 2019 Karol Lassak
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import operator
+
+from datetime import datetime, timedelta
+
+from c7n.utils import chunks, local_session, type_schema, get_annotation_prefix
+from c7n.filters import FilterValidationError, Filter
+
+from c7n_gcp.provider import resources as gcp_resources
+
+
+class MetricsFilter(Filter):
+    """
+
+    Filters GCP resources based on live metrics from the Stackdriver monitor
+
+    Click `here
+    <https://cloud.google.com/monitoring/api/metrics/>`_
+    for a full list of metrics supported by GCP resources.
+
+    :example:
+
+    Find all VMs with an average Percentage CPU greater than 75% over last 2 hours
+
+    .. code-block:: yaml
+
+        policies:
+          - name: vm-percentage-cpu
+            resource: azure.vm
+            filters:
+              - type: metric
+                metric: Percentage CPU
+                aggregation: average
+                op: gt
+                threshold: 75
+                timeframe: 2
+
+    :example:
+
+    Find KeyVaults with more than 1000 API hits in the last hour
+
+    .. code-block:: yaml
+
+        policies:
+          - name: keyvault-hits
+            resource: azure.keyvault
+            filters:
+              - type: metric
+                metric: ServiceApiHit
+                aggregation: total
+                op: gt
+                threshold: 1000
+                timeframe: 1
```

Comment:
afaics in the api docs the time frame has configurable units in the api based on the trailing suffix, we should expose this ideally to the user, so they can express in the units most natural to them.